kernel/Kconfig

```diff
@@ -1006,6 +1006,20 @@ config STATIC_INIT_GNU
 	  Note the option CMAKE_LINKER_GENERATOR does not yet support this feature
 	  or CPP.
 
+config BOOTARGS
+	bool "Support bootargs"
```

Comment:
Technically speaking, bootargs can always be enabled. In the static case, just default the args to `""` and in the dynamic case, just default the size to 1 (and ensure that the storage is static).

Reply:
> Technically speaking, bootargs can always be enabled

That's true, however:

If we keep the bootargs always enabled, `main(int, char **)` will be called unconditionally. Even if it will be linked with `main(void)` later on. White this seems to work just fine (at least on x86, for which it was tested), I fear that this will be prone to randomly break, especially when LTO is enabled.

Plus, the bootargs feature bring some new code that the user might not want to always include. So I'd say - let's leave that choice to the user.